mne/io/fiff/tests/test_raw.py

```diff
@@ -115,6 +115,19 @@ def test_copy_append():
     assert_equal(data.shape[1], 2 * raw._data.shape[1])
 
 
+@testing.requires_testing_data
+def test_append_channels():
+    """Test raw splitting / re-appending channel types
+    """
+    raw = Raw(fif_fname, preload=True)
+    raw_eeg = raw.pick_types(eeg=True, meg=False)
+    raw_meg = raw.pick_types(meg=True, eeg=False)
+    raw_new = raw_meg.append_channels([raw_meg])
+    assert_true([ch in raw_new.ch_names for ch in raw.ch_names])
+    assert_true(raw_new.n_times == raw.n_times)
+    assert_true(len(raw_new.ch_names) == len(raw.ch_names))
```

Comment:
you should actually check that `raw[:366, :]` times and data are equal to `raw_new[:, :]`